deployments/hardhat/dai/deploy.ts

```diff
@@ -16,43 +9,47 @@ async function makeToken(
   decimals: number,
   symbol: string
 ): Promise<FaucetToken> {
-  return await deploymentManager.deploy<
-    FaucetToken,
-    FaucetToken__factory,
-    [string, string, number, string]
-  >('test/FaucetToken.sol', [
-    (BigInt(amount) * 10n ** BigInt(decimals)).toString(),
-    name,
-    decimals,
-    symbol,
-  ]);
+  const mint = (BigInt(amount) * 10n ** BigInt(decimals)).toString();
+  return deploymentManager.deploy(symbol, 'test/FaucetToken.sol', [mint, name, decimals, symbol]);
```

Comment:
This is one of the biggest open questions for me (you'll see several XXXs about api of return/roots) but I wanted to come back to it after I had everything else working

Reply:
Yeah, it's not huge and it would be nice if there were an auto-type for it. I wonder maybe if we could use a little TypeScriptFu to make it a bit better, like a type that defines sub-types so it's just a deploy but it auto-discovers the correct set of types. Happy to help here and/or we could ask @fluffywaffles

Reply:
My question is less about the type and more about when we should be writing roots and who should supply the addresses